### DISCUSSION TITLE
Moving to M1 (Apple Silicon) procedure using Time Machine/clone

### DISCUSSION THREAD

**Discussion by jameslbrown** [Homebrew/discussions#2039] [Everyday usage]
Had to spend a bit of time going through all the discussions, so thought I'd put it what worked for me, to save other people time.

After you've used Time Machine/Migration Assistant or cloned using Carbon Copy Cloner /SuperDuper! etc on new Apple Silicon machine:

cd to ~/Documents or somewhere you want to temporarily store your list of formulae.

`brew update
brew bundle dump
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

then issue the _echo_ and _eval_ commands it tells you to, finally:

 `brew bundle --file Brewfile`

everything should be installed as Apple Silicon native if they are available. You'll be informed if they are not.

Hope this helps someone.

**Comment by SMillerDev** [comment]
You really shouldn't include brew in your time machine backups. It's a better idea to run `brew bundle` on the old machine and just migrate the `Brewfile` and possibly `/usr/local/etc`
